src/envoy/auth/controller.cc

```diff
@@ -0,0 +1,175 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "controller.h"
+
+namespace Envoy {
+namespace Http {
+namespace Auth {
+namespace {
+
+// The authorization HTTP header.
+const LowerCaseString kAuthorizationKey("authorization");
+
+// The autorization bearer prefix.
+const std::string kBearerPrefix = "Bearer ";
+
+// The HTTP header to pass verified token payload.
+const LowerCaseString kJwtPayloadKey("sec-istio-auth-userinfo");
+
+// The per-request JWT authentication object.
+class AuthRequest : public Logger::Loggable<Logger::Id::http>,
```

Comment:
Can we name JwtAuthRequest, in case we add other authentication object in future

Reply:
Other Authentication may use this class too.  The key is it is pre-request data